src/test/spec/unified_runner/operation.rs

```diff
@@ -93,16 +103,104 @@ pub trait TestOperation: Debug + Send + Sync {
     }
 }
 
+macro_rules! with_mut_session {
```

Comment:
to facilitate working with sessions through the lock, this macro pops it out of the entity map, "passes" it to the provided block, and then returns it to the entity map. It does it this way so that we can continue to borrow the entity map in other ways even when we're using a session, which we'd have to borrow mutably from the map.

Reply:
nice solution. might be worth a comment explaining why we need it?